sway-fmt-v2/src/utils/attributes.rs

```diff
@@ -1,11 +1,82 @@
-use crate::Formatter;
-use sway_parse::AttributeDecl;
+use crate::fmt::{Format, FormattedCode, Formatter};
+use sway_parse::{
+    attribute::{Annotated, AttributeDecl},
+    token::Delimiter,
+    Parse,
+};
 use sway_types::Spanned;
-pub fn format_attributes(attributes: Vec<AttributeDecl>, _formatter: &mut Formatter) -> String {
-    // TODO format attributes
-    attributes
-        .into_iter()
-        .map(|x| x.span().as_str().to_string())
-        .collect::<Vec<String>>()
-        .join("\n")
+
+use super::bracket::{Parenthesis, SquareBracket};
+
+impl<T: Parse> Format for Annotated<T> {
+    fn format(&self, formatter: &mut Formatter) -> FormattedCode {
+        let attributes = &self.attribute_list;
+        let mut formatted_code = String::new();
+
+        for attr in attributes {
+            AttributeDecl::format(attr, &mut formatted_code, formatter);
+        }
+
+        formatted_code
```

Comment:
Should we also format `T` here? E.g. do we want to change `T: Parse` to `T: Parse + Format` and do something like:

```suggestion
        formatted_code + self.value.format(formatter)
```

Or is this being handled separately?

Reply:
I wasn't sure if we should include item formatting in the scope of a function intended to format annotations. Lmk if we should do this and I'll move that logic here, it's still in fmt.rs

Reply:
Also would like to tackle that in a separate PR if we intend to do it

Reply:
Went ahead and did it, I think it does feel cleaner after all

Reply:
> I wasn't sure if we should include item formatting in the scope of a function intended to format annotations.

I guess it's worth keeping in mind that when we say `impl` `Format` for `Annotated<T>`, we are actually saying that this implementation applies to all annotated `T`s, not just the annotations themselves. I think this turned out nice :ok_hand: